Explorer/Assets/DCL/UI/GenericContextMenu/Controllers/Communities/CommunityInvitationContextMenuButtonHandler.cs

```diff
@@ -0,0 +1,130 @@
+        /// when the user hovers the submenu button.
+        /// </summary>
+        /// <param name="contextMenu">Any context menu to which to add the button.</param>
+        /// <param name="buttonText">The text to show in the new button.</param>
+        /// <param name="buttonIcon">The icon to show next to the new button.</param>
+        public void AddSubmenuControlToContextMenu(GenericContextMenuParameter.GenericContextMenu contextMenu, string buttonText, Sprite buttonIcon)
+        {
+            contextMenu.AddControl(new SubMenuContextMenuButtonSettings(buttonText,
+                                                                        buttonIcon,
+                                                                        new GenericContextMenuParameter.GenericContextMenu(contextMenu.width,
+                                                                                         verticalLayoutPadding: contextMenu.verticalLayoutPadding,
+                                                                                         elementsSpacing: contextMenu.elementsSpacing,
+                                                                                         offsetFromTarget: contextMenu.offsetFromTarget),
+                                                                        asyncControlSettingsFillingDelegate: CreateInvitationSubmenuItemsAsync,
+                                                                        asyncVisibilityResolverDelegate: ResolveInvitationSubmenuVisibilityAsync));
+        }
+
+        /// <summary>
+        /// Replaces the remote user to which invitations will be sent when possible.
+        /// </summary>
+        /// <param name="walletId">The new wallet address.</param>
+        public void SetUserToInvite(string walletId)
+        {
+            userWalletId = walletId;
+        }
+
+        private async UniTask<bool> ResolveInvitationSubmenuVisibilityAsync(CancellationToken ct)
+        {
+            if(ct.IsCancellationRequested)
+                return false;
+
+            // Asks the server for the data of the communities to which the user can be invited
+            Result<GetInvitableCommunityListResponse> response = await communitiesDataProvider.GetInvitableCommunityListAsync(userWalletId, ct).SuppressToResultAsync(ReportCategory.COMMUNITIES);
+
+            if (response.Success)
+            {
+                lastCommunityData = response.Value.data;
+                lastCommunityNames.Clear();
+
+                foreach (GetInvitableCommunityListResponse.InvitableCommunityData communityData in lastCommunityData)
+                    lastCommunityNames.Add(communityData.name);
+
+                if(lastCommunityNames.Count > 0)
+                    return true;
+            }
+            else
+            {
+                // TODO
```

Comment:
Is this TODO intended to be implemented later in the hardening? I guess here we should show a warning with the error in the same way we do for the rest of calls to the backend, right?